internal/agent/device/status/resources.go

```diff
@@ -0,0 +1,43 @@
+package status
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/agent/device/resource"
+)
+
+var _ Exporter = (*Resources)(nil)
+
+type Resources struct {
+	manager resource.Manager
+}
+
+func newResources(manager resource.Manager) *Resources {
+	return &Resources{
+		manager: manager,
+	}
+}
+
+func (r *Resources) Export(ctx context.Context, status *v1alpha1.DeviceStatus) error {
+	usage, err := r.manager.Usage()
+	if err != nil {
+		status.Resources.Disk = v1alpha1.DeviceResourceStatusUnknown
+		return fmt.Errorf("getting resource usage: %w", err)
+	}
+
+	switch {
+	case usage.FsUsage.IsAlert():
+		status.Resources.Disk = v1alpha1.DeviceResourceStatusCritical
```

Comment:
1. Do we want to standardize on disk vs file system?
2. Do we have any way to add a message here to say which file system is reaching capacity and how much is left?

Reply:
Data dictionary provided more context in the degraded state. So we can push more data in warn and alert.

Reply:
Also, it makes sense to add an error state to the enum. Will update.